src/librustc_target/spec/mod.rs

```diff
@@ -635,6 +635,12 @@ pub struct TargetOptions {
     /// typically because the platform needs to unwind for things like stack
     /// unwinders.
     pub requires_uwtable: bool,
+
+    /// Targets like AMDGPU add alwaysinline hints everywhere,
+    /// and basically inline everything regardless of our wishes anyway.
+    /// This causes LLVM validation errors due to the conflicting attributes.
+    /// Defaults to false.
+    pub ignore_inline_never: bool,
```

Comment:
General strategy is to have `is_like_platform` for platform-specific handling.

Reply:
This seems fine to me, though.

Reply:
Please don't do this, if it's a LLVM-specific *temporary hack*, add code for it checking e.g. architecture name in triple, to `rustc_codegen_llvm`.